userspace/libsinsp/container.cpp

```diff
@@ -78,9 +78,10 @@ bool sinsp_container_manager::remove_inactive_containers()
 		{
 			if(containers_in_use.find(it->first) == containers_in_use.end())
 			{
+				std::shared_ptr<sinsp_container_info> container = it->second;
 				for(const auto &remove_cb : m_remove_callbacks)
 				{
-					remove_cb(m_containers[it->first]);
+					remove_cb(*container);
```

Comment:
MAJOR
I thought a big point of this was to make the call backs async potentially. So don't you want these to take a shared_ptr<const container_info> ?

Reply:
One commit at a time :) But since they get passed a reference, they can't e.g. store the pointer somewhere and while they're running the shared_ptr is kept alive by the local variable created just above.

Reply:
Yeah, we don't need to solve this now. 

The worry is if the client does anything async with the object. We could either pass shared_ptr<const x> &obj or weak_ptr<x> obj so they had the option to take partial ownership.

Reply:
Let's burn that bridge when we get there ;) Also I think that this applies more to the new container callback rather than the remove one.